### PR TITLE
Add secure Redis compose file

### LINKED ISSUE #225: Need for running the security services with Redis database
Do we need to add docker-compose-redis.yml and docker-compose-redis-arm64.yml in nightly-build for running the security services with Redis?

### PR DESCRIPTION
**Pull request by andresrinivasan**
Fixes #225 

This compose file is tightly coupled to the related PRs on edgex-go and go-mod-boostrap.

This is tightly coupled to 
[go-mod-bootstrap PR#68](https://github.com/edgexfoundry/go-mod-bootstrap/pull/68)
[edgex-go PR#2497](https://github.com/edgexfoundry/edgex-go/pull/2497)

Signed-off-by: Andre Srinivasan <andre@redislabs.com>

### CODE REVIEW

**Comment by tonyespy** [comment]
@andresrinivasan Can you add a comment with links to the edgex-go and go-mod-bootstrap PRs?

File: releases/nightly-build/compose-files/docker-compose-nexus-redis-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,574 @@
+# /*******************************************************************************
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-redis-arm64.yml:1]: Will review the non-arm64 instead.  Most comments there will apply here.

File: releases/nightly-build/compose-files/docker-compose-nexus-redis.yml [hunk 2]
```diff
@@ -0,0 +1,574 @@
+# /*******************************************************************************
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:1]: Please rebase on top of upstream/master.  Will try to note specifically the changes that need to be picked up.

File: releases/nightly-build/compose-files/docker-compose-nexus-redis.yml [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+# all common shared environment variables defined here:
+x-common-env-variables: &common-variables
+  Registry_Host: edgex-core-consul
+  Clients_CoreData_Host: edgex-core-data
+  Clients_Logging_Host: edgex-support-logging
+  Clients_Notifications_Host: edgex-support-notifications
+  Clients_Metadata_Host: edgex-core-metadata
+  Clients_Command_Host: edgex-core-command
+  Clients_Scheduler_Host: edgex-support-scheduler
+  Logging_EnableRemote: "true"
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:29]:
```suggestion
  Clients_RulesEngine_Host: edgex-kuiper
  Clients_VirtualDevice_Host: edgex-device-virtual
```

File: releases/nightly-build/compose-files/docker-compose-nexus-redis.yml [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+  vault-config:
+  vault-file:
+  vault-logs:
+  # non-shared volumes
+  portainer_data:
+  secrets-setup-cache:
+
+services:
+  volume:
+    image: nexus3.edgexfoundry.org:10004/docker-edgex-volume:master
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:57]: Trying to get rid of edgex-volume.

```suggestion
    image: ubuntu:latest
    entrypoint: [ "tail", "-f", "/dev/null" ]

```

File: releases/nightly-build/compose-files/docker-compose-nexus-redis.yml [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+
+  consul:
+    image: nexus3.edgexfoundry.org:10004/docker-edgex-consul:master
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+    container_name: edgex-core-consul
+    hostname: edgex-core-consul
+    environment:
+      - "SECRETSTORE_SETUP_DONE_FLAG=/tmp/edgex/secrets/edgex-consul/.secretstore-setup-done"
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:75]:
```suggestion
      - "SECRETSTORE_SETUP_DONE_FLAG=/tmp/edgex/secrets/edgex-consul/.secretstore-setup-done"
      - EDGEX_DB=mongo
      - EDGEX_SECURE=true
```

Comment (andresrinivasan) [reply releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:75]: Do you mean `EDGEX_DB=redis`?

Comment (bnevis-i) [reply releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:75]: Yes.

Comment (andresrinivasan) [reply releases/nightly-build/compose-files/docker-compose-nexus-redis.yml:75]: Done

**Comment by andresrinivasan** [comment]
Apologies but I managed to muck up my branch and in recovering, hosed the PR. Opening a new PR.

**Pull request closed without merging**